_core/smart/tutorial-counter.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: core
+description: "Blockstack Clarity: Counter Tutorial"
+permalink: /:collection/:path.html
+---
+# Tutorial: Counter
+
+| Experience | | **Intermediate**  |
+| Duration | | **30 minutes** |
+
+In this tutorial, you learn how to implement a smart contract that stores and manipulates an integer value on the Stacks 2.0 Blockchain. By the end of this tutorial, you will ...
+
+* Have experienced test-driven development with Clarity
+* Understand more Clarity language design principles
+* Have a working Clarity counter smart contract
+
+## Overview
+
+* TOC
+{:toc}
+
+## Pre-requisites
+
+Before you get started, you should complete the [Hello World tutorial](tutorial.html).
+
+## Step 1: Download counter tutorial project
+
+In this step, you initialize a starter project with additional counter tutorial files:
+
+1. Using your terminal, run the following command:
+
+    ```bash
+    npm init clarity-starter
+    ```
+
+    You have to select a template and a name for your local folder. For the counter template used in this tutorial, ensure to type `counter` and hit ENTER:
+
+    ```bash
+    ? Template - one of [hello-world, counter]: counter
+    ? Project name: (clarity-counter-project)
+    ```
+
+    Finally, the project dependencies are installed and your project is ready for development. Because you already completed the Hello World tutorial, the project structure is familiar to you. The main difference is that we have additional tests for a new counter smart contract.
+
+## Step 2: Running tests
+
+Smart contracts are often developed in a test-driven approach to ensure code quality but also to speed up the development cycle by removing the need to push every change to the Blockchain before executing it. We will do the same in this project. Now, let's run the tests and review the results:
+
+1. Still in the project root directory, run the following command:
+
+    ```bash
+    npm test
+    ```
+
+    You should see the following response:
+
+    ```bash
+    counter contract test suite
+        ✓ should have a valid syntax
+        deploying an instance of the contract
+        1) should start at zero
+        2) should increment
+        3) should decrement
+
+    hello world contract test suite
```

Comment:
Need to remove this section. I think this is coming from a local existing package, because the "hello world" tests should not be there. The test results should be "1 passing, 3 failing"